src/spec_procs.c

```diff
@@ -697,3 +698,66 @@ SPECIAL(bank)
     return (FALSE);
 }
 
+
+void do_spells_or_skills(int mode, struct char_data *ch)
```

Comment:
I am not an expert, but I dont think this code is in the right place.

Maybe its a good idea to reconsider the do_spells_or_skills and refactor it in two (or even three) distinct methods? The approach to change the method behavior based on an argument don't look good to me.